qiskit/ignis/measurement/discriminator/discriminators.py

```diff
@@ -0,0 +1,426 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+from abc import abstractmethod
+from typing import Union, List
+
+import numpy as np
+from sklearn.discriminant_analysis import LinearDiscriminantAnalysis
+from sklearn.discriminant_analysis import QuadraticDiscriminantAnalysis
+from sklearn.preprocessing import StandardScaler
+
+from qiskit.exceptions import QiskitError
+from qiskit.pulse import PulseError
+from qiskit.result import Result
+from qiskit.pulse.schedule import Schedule
+
+
+class BaseDiscriminationFitter:
+    """
+    IQDiscriminatorFitter takes IQ level 1 data produced by calibration
+    measurements with a known expected state. It fits a discriminator
+    that can be used to produce level 2 data, i.e. counts of quantum states.
+    """
+    @abstractmethod
```

Comment:
Should this be abstract?

Reply:
Does not have to be. I realized however, that I forgot to have `BaseDiscriminationFitter` inherit from `ABC` to prevent the creation of `BaseDiscriminationFitter` and `IQDiscriminationFitter` objects. Only `LinearIQDiscriminator` and `QuadraticIQDiscriminator` can be initialized. I fixed this in 0395d50.